spec/requests/internal/users_spec.rb

```diff
@@ -66,35 +64,50 @@ def full_profile
   context "when merging users" do
     before do
       full_profile
-      post "/internal/users/#{user.id}/merge", params: { user: { merge_user_id: user2.id } }
-      Delayed::Worker.new(quiet: true).work_off
     end
 
     it "deletes duplicate user" do
+      post "/internal/users/#{user.id}/merge", params: { user: { merge_user_id: user2.id } }
+
       expect { User.find(user2.id) }.to raise_exception(ActiveRecord::RecordNotFound)
     end
 
     it "merges all content" do
-      expect(user.comments.count).to eq(2)
-      expect(user.articles.count).to eq(2)
-      expect(user.reactions.count).to eq(4)
+      expected_articles_count = user.articles.count + user2.articles.count
```

Comment:
using explicit conditions before the HTTP request, to make sure a randomized test order wouldn't break these